presto-main/src/main/java/com/facebook/presto/connector/ConnectorManager.java

```diff
@@ -248,6 +248,8 @@ private synchronized void addConnector(String catalogName, String connectorId, C
         if (indexResolver != null) {
             indexManager.addIndexResolver(connectorId, indexResolver);
         }
+
+        metadataManager.getSessionPropertyManager().addConnectorSessionProperties(catalogName, connector.getSessionProperties());
```

Comment:
If this isn't allowed to be null, fetch the value above with a check, like the others